alicloud/resource_alicloud_drds_instance_test.go

```diff
@@ -0,0 +1,252 @@
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func init() {
+	resource.AddTestSweepers("alicloud_drds_instance", &resource.Sweeper{
+		Name: "alicloud_drds_instance",
+		F:    testSweepDRDSInstances,
+	})
+}
+
+func testSweepDRDSInstances(region string) error {
+	rawClient, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting Alicloud client: %s", err)
+	}
+	client := rawClient.(*connectivity.AliyunClient)
+
+	prefixes := []string{
+		"tf-testAcc",
+		"tf_testAcc",
+	}
+
+	var insts []drds.Instance
+	req := drds.CreateDescribeDrdsInstancesRequest()
+	req.RegionId = client.RegionId
+	for {
+		raw, err := client.WithDrdsClient(func(drdsClient *drds.Client) (interface{}, error) {
+			return drdsClient.DescribeDrdsInstances(req)
+		})
+		if err != nil {
+			return fmt.Errorf("Error retrieving DRDS Instances: %s", err)
+		}
+		resp, _ := raw.(*drds.DescribeDrdsInstancesResponse)
+		if resp == nil || len(resp.Data.Instance) < 1 {
+			break
+		}
+		insts = append(insts, resp.Data.Instance...)
+
+	}
+
+	sweeped := false
+	for _, v := range insts {
+		name := v.Description
+		id := v.DrdsInstanceId
+		skip := true
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(strings.ToLower(name), strings.ToLower(prefix)) {
+				skip = false
+				break
+			}
+		}
+		if skip {
+			log.Printf("[INFO] Skipping DRDS Instance: %s (%s)", name, id)
+			continue
+		}
+
+		sweeped = true
+		log.Printf("[INFO] Deleting DRDS Instance: %s (%s)", name, id)
+		req := drds.CreateRemoveDrdsInstanceRequest()
+		req.DrdsInstanceId = id
+		_, err := client.WithDrdsClient(func(drdsClient *drds.Client) (interface{}, error) {
+			return drdsClient.RemoveDrdsInstance(req)
+		})
+		if err != nil {
+			log.Printf("[ERROR] Failed to delete DRDS Instance (%s (%s)): %s", name, id, err)
+		}
+	}
+	if sweeped {
+		// Waiting 30 seconds to eusure these DB instances have been deleted.
+		time.Sleep(30 * time.Second)
+	}
+	return nil
+}
+
+func TestAccAlicloudDRDSInstance_Basic(t *testing.T) {
+	var instance drds.DescribeDrdsInstanceResponse
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		IDRefreshName: "alicloud_drds_instance.basic",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckDRDSInstanceDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccDrdsInstance,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDRDSInstanceExist(
+						"alicloud_drds_instance.basic", &instance),
+					resource.TestCheckResourceAttr(
+						"alicloud_drds_instance.basic",
+						"pay_type",
+						"Postpaid"),
+					resource.TestCheckResourceAttr(
+						"alicloud_drds_instance.basic",
+						"instance_series",
+						"drds.sn1.4c8g"),
+					resource.TestCheckResourceAttr(
+						"alicloud_drds_instance.basic",
+						"zone_id",
+						"cn-hangzhou-b"),
```

Comment:
If region is not cn-hangzhou, line 112 will be error. Please use `resource.TestCheckResourceAttrSet("alicloud_drds_instance.basic","zone_id")` instead line 112-115.